src/core/net/netif.hpp

```diff
@@ -156,6 +157,41 @@ class NetifMulticastAddress : public otNetifMulticastAddress,
     bool Matches(const Address &aAddress) const { return GetAddress() == aAddress; }
 };
 
+class ExternalNetifMulticastAddress : public NetifMulticastAddress
+{
+    friend class Netif;
+    friend class LinkedList<ExternalNetifMulticastAddress>;
+
+public:
+#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_MLR_ENABLE
```

Comment:
is `(OPENTHREAD_FTD || OPENTHREAD_MTD)` necessary?

Reply:
removed